public/app/plugins/datasource/prometheus/configuration/PromSettings.tsx

```diff
@@ -67,14 +68,23 @@ export const PromSettings = (props: Props) => {
           </InlineFormLabel>
           <Select
             options={httpOptions}
-            value={httpOptions.find(o => o.value === value.jsonData.httpMethod)}
-            onChange={onChangeHandler('httpMethod', value, onChange)}
+            value={httpOptions.find(o => o.value === options.jsonData.httpMethod)}
+            onChange={onChangeHandler('httpMethod', options, onOptionsChange)}
             width={7}
           />
         </div>
       </div>
       <h3 className="page-heading">Misc</h3>
       <div className="gf-form-group">
+        <div className="gf-form">
+          <Switch
+            checked={options.jsonData.disableLanguageProvider}
+            label="Disable metric lookup"
+            labelClass="width-14"
+            onChange={onUpdateDatasourceJsonDataOptionChecked(props, 'disableLanguageProvider')}
+            tooltip="Switch on if you experience performance issues in the query editor. This will disable autocomplete and the metrics chooser."
```

Comment:
Should be clarified to only disable the metric and label value lookups.

Reply:
But how can the metrics chooser work without metrics data? (Might be good if you suggested a concrete wording here because I'm not sure whether I understood your comment correctly.)

Reply:
Yeah, we need to make clear that the metrics chooser will be off/disabled/not working if the user enables this option.

Reply:
"Checking this option will disable the Metrics chooser and metric/label support in the query field's tab-completion. This helps if you have performance issues with bigger Prometheus instances."